### ISSUE TITLE
[Correctness] Maybe functions get incorrectly translated

### ISSUE THREAD

**Issue by sstur** [Khan/flow-to-ts#197]
The following ([see here](https://flow-to-ts.netlify.app/#code=dHlwZSBGb28gPSB7CiAgbmFtZTogc3RyaW5nLAogIG9uQ2hhbmdlPzogPyhzdHJpbmcpID0+IHZvaWQsCn07Cg==&prettier=1&semi=1&singleQuote=0&tabWidth=2&trailingComma=all&bracketSpacing=0&arrowParens=always&printWidth=80&inlineUtilityTypes=0)):
```js
type Foo = {
  name: string,
  onChange?: ?(string) => void,
};
```
Gets incorrectly translated to:
```ts
type Foo = {
  name: string;
  onChange?: (arg0: string) => void | null | undefined;
};
```
The problem is related to operator precedence. The arrow function should to be wrapped in parens:
```ts
type Foo = {
  name: string;
  onChange?: ((arg0: string) => void) | null | undefined;
};
```

**Comment by k4b7** [comment]
Part of me wonders if it'd be easier to convert `?T` to `Maybe<T>` and then provide a `Maybe` type in a global.d.ts.

**Comment by k4b7** [comment]
This fix for this is the v0.2.2 release.  I decided not to go with `Maybe<T>` because we aren't generating a global.d.ts.  I'd like to refactor the CLI interface to have separate commands to convert files, generate new files (e.g. global.d.ts, tsconfig.json, etc.), and update other non-js files (e.g. package.json, .eslintrc.js, etc.).  Once that's in place it'll be easier to adopt `Maybe<T>`.
